ludwig/combiners/combiners.py

```diff
@@ -514,28 +570,63 @@ def output_shape(self) -> torch.Size:
         return self.tabnet.output_shape
 
 
+# TODO(ksbrar): Refactor transformers into using base class for common attrs?
 @dataclass
 class TransformerCombinerConfig:
     num_layers: int = schema.PositiveInteger(default=1)
+    """TODO (default: 1)."""
+
     hidden_size: int = schema.NonNegativeInteger(default=256)
+    """The size of the hidden representation within the transformer block. It is usually the same as `embedding_size`,
+       but if the two values are different, a projection layer will be added before the first transformer block.
+       (default: 256)."""
+
     num_heads: int = schema.NonNegativeInteger(default=8)
+    """Number of heads of the self attention in the transformer block (default: 8)."""
+
     transformer_output_size: int = schema.NonNegativeInteger(default=256)
+    """Size of the fully connected layer after self attention in the transformer block. This is usually the same as
+       `hidden_size` and `embedding_size` (default: 256)."""
+
     dropout: float = schema.FloatRange(default=0.1, min=0, max=1)
+    """Dropout rate for the transformer block (default: 0.1)."""
+
     fc_layers: Optional[List[Dict[str, Any]]] = schema.DictList()
     num_fc_layers: int = schema.NonNegativeInteger(default=0)
+    """The number of stacked fully connected layers (only applies if `reduce_output` is not null) (default: 0)."""
```

Comment:
In a future PR, want to add some more logic for validating conditional fields like this one

Reply:
https://github.com/ludwig-ai/ludwig/issues/1736 - add this to the comment